### PR TITLE
Andrew features..

### PR DESCRIPTION
**Pull request by epcim**
give a try, fixes needed

### CODE REVIEW

File: reclass/version.py [hunk 1]
```diff
@@ -7,12 +7,12 @@
...
+DESCRIPTION = 'merge data by recursive descent down an ancestry hierarchy (forked extended version)'
+VERSION = '1.5.2'
+AUTHOR = 'martin f. krafft / Andrew Pickford'
+AUTHOR_EMAIL = 'andrewp@nikhef.nl'
+MAINTAINER = 'Andrew Pickford'
+MAINTAINER_EMAIL = ''
+COPYRIGHT = 'Copyright © 2007–14 martin f. krafft, extensions © 2017 Andrew Pickford'
 LICENCE = 'Artistic Licence 2.0'
-URL = 'https://github.com/madduck/reclass'
+URL = 'https://github.com/AndrewPickford/reclass'
```
Comment (epcim) [inline reclass/version.py]: change url here

File: README.extentions [hunk 2]
```diff
@@ -0,0 +1,205 @@
+Escaping of References and Inventory Queries
```
Comment (epcim) [inline README.extentions]: lgtm, nice!

File: README.extentions [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+parameters:   
+  colour: Blue
+  unescaped: The colour is Blue   
+  escaped: The colour is ${colour}   
+  double_escaped: The colour is \Blue
+
+
+
+Merging Referenced Lists and Dictionaries
```
Comment (epcim) [inline README.extentions]: lgtm

File: README.extentions [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+  two:
+    c: 3
+    d: 4
+
+This first sets the parameter three to the value of parameter one (class test1) then merges parameter two into
+parameter three (class test2) and finally merges the parameter three definition given in the node definition into
+
+
+
+Nested References
```
Comment (epcim) [inline README.extentions]: lgtm, 
just interested what will be reported as missing key?

File: README.extentions [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+    two: a
+  beta:
+    a: 99
+
+The ${beta:${alpha:two}} construct first resolves the ${alpha:two} reference to the value 'a', then resolves
+the reference ${beta:a} to the value 99.
+
+
+
+Inventory Queries
```
Comment (epcim) [inline README.extentions]: lgtm
I think we need a real world example here.

File: README.rst [hunk 6]
```diff
@@ -1,5 +1,31 @@
-reclass README
-==============
+reclass (extended) README
+=========================
 
-The documentation for **reclass** is available from
-http://reclass.pantsfullofunix.net.
+This is an extended version of **reclass** available at:
+https://github.com/AndrewPickford/reclass
```
Comment (epcim) [inline README.rst]: I would only state That README.extensions.rst contains features added above the reclass.pantsfullofunix.net version.

We might tend to ship this for as package one day? As well as publish it's generated doc, so link should go here as well.

File: README.rst [hunk 7]
```diff
@@ -1,5 +1,31 @@
...
+
+The escaping of references changes how the constructs '\${xxx}' and '\\${xxx}'
+are rendered.
+
+
+Documentation
+=============
+
+Documentation covering the original version is in the doc directory.
+See the README.extensions file for documentation on the extentions.
```
Comment (epcim) [inline README.rst]: Link to generated doc at https://salt-formulas.readthedocs.io/ 
README.extensions.rst

File: README.rst [hunk 8]
```diff
@@ -1,5 +1,31 @@
...
+Extentions
+==========
+
+Escaping of References and Inventory Queries
+Merging Referenced Lists and Dictionaries
+Nested References
+Inventory Queries
+
+
+Changes
```
Comment (epcim) [inline README.rst]: we might add generated CHANGELOG,
or use https://github.com/openstack/reno

File: README.extentions (hunk 4 above)
Comment (AndrewPickford) [reply README.extentions]: Given:
```yaml
# class /srv/salt/env/pre-prod/classes/role/server/dns/init.yml
# (node vaars-09)
parameters:
  lookup:
    a: 1
    b: 2
  lookup_key: c
  test: ${lookup:${lookup_key}}
```
The error reported is:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/reclass/values/refitem.py", line 53, in _resolve
    return path.get_value(context)
  File "/usr/lib/python2.7/site-packages/reclass/utils/dictpath.py", line 128, in get_value
    return self._get_innermost_container(base)[self._get_key()]
KeyError: 'c'

-> vaars-09.nikhef.nl
   Cannot resolve ${lookup:c}, at test, in yaml_fs:///srv/salt/env/pre-prod/classes/role/server/dns/init.yml

**Comment by epcim** [comment]
Closing in favor of rebased version: https://github.com/salt-formulas/reclass/pull/8

**Comment by epcim** [comment]
@AndrewPickford I have send your a msg over keybase (https://keybase.io/docs/extension)

```
I rebased your reclass master branch against https://github.com/salt-formulas/reclass (master) and we merged the result into the "develop" branch.

Within a week or two we will verify this  "develop" in our CI and will merge it into master.

The actual diff is here (however github seems to be confused):
https://github.com/salt-formulas/reclass/pull/9#issuecomment-329089756

The actual rebase is here (+ additional doc/readme update):
https://github.com/salt-formulas/reclass/pull/8

I am sorry for the inconvenience with these diffs. Basically, it's just Doc + Ingore class notfound made better.
```

**Pull request closed without merging**